PWGCF/Flow/Tasks/flowPtEfficiency.cxx

```diff
@@ -89,8 +146,14 @@ struct flowPtEfficiency {
   }
   PROCESS_SWITCH(flowPtEfficiency, processReco, "process reconstructed information", true);
 
-  void processSim(aod::McCollision const&, soa::SmallGroups<soa::Join<aod::McCollisionLabels, aod::Collisions>> const& collisions, myMcParticles const& mcParticles)
+  void processSim(myMcCollisions::iterator const& collision, aod::BCsWithTimestamps const&, soa::SmallGroups<soa::Join<aod::McCollisionLabels, aod::Collisions>> const& collisions, myMcParticles const& mcParticles)
   {
+    if (cfgSelRunNumberEnabled) {
+      auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+      int RunNumber = bc.runNumber();
+      if (!std::count(cfgRunNumberList.value.begin(), cfgRunNumberList.value.end(), RunNumber))
```

Comment:
Same here
Also consider that the run number should be the same at detector level as at generator level

Reply:
Hi Victor, thank you for your comments! However, I have one question: how can I check the run number at the generator level? I temporarily can't find any methods in the AO2D tables or in the previous code to achieve this. Could you please provide some guidance?

Reply:
Well, I guess you are doing it in here
This is your generator level process function, right?
You are getting your generator level collision which brings the `bc` from which you extract the run number out of the `BCWithTimestamps` table
Are you still missing something?